include/boost/astronomy/io/default_hdus.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef BOOST_ASTRONOMY_IO_DEFAULT_HDUS_HPP
+#define BOOST_ASTRONOMY_IO_DEFAULT_HDUS_HPP
+
+
+#include<boost/astronomy/io/hdu.hpp>
+#include<boost/astronomy/io/primary_hdu.hpp>
+#include<boost/astronomy/io/binary_table.hpp>
+#include<boost/astronomy/io/ascii_table.hpp>
+#include<boost/variant.hpp>
+
+
+
+namespace boost { namespace astronomy {namespace io {
+    struct default_hdu_manager{
+
+        typedef boost::variant<
+            boost::blank,
+            primary_hdu,   
+            binary_table_extension,
+            ascii_table
+
+        >  Extension;
+
+        static Extension construct_hdu(header& hdu_header, const std::string& data_buffer = "") {
+            if (hdu_header.contains_keyword("SIMPLE")) {
```

Comment:
This check will go now as there is a particular member function for getting name of eextension